hs_core/views/__init__.py

```diff
@@ -69,6 +72,38 @@ def add_file_to_resource(request, shortkey, *args, **kwargs):
     extract_metadata = request.REQUEST.get('extract-metadata', 'No')
     extract_metadata = True if extract_metadata.lower() == 'yes' else False
 
+    irods_fname = request.session.get('irods_add_file_name', '')
+    if irods_fname:
+        res_cls = resource.__class__
+        file_types = res_cls.get_supported_upload_file_types()
+        valid = False
+        if file_types == ".*":
+            valid = True
+        else:
+            ext = os.path.splitext(irods_fname)[1]
+            if ext == file_types:
+                valid = True
+            else:
+                for index in range(len(file_types)):
+                    if ext == file_types[index].strip():
+                        valid = True
+                        break
+        if not valid:
+            request.session['file_type_error'] = "Invalid file type: {ext}".format(ext=ext)
+            return HttpResponseRedirect(request.META['HTTP_REFERER'])
+        else:
+            user = request.session["user"]
+            password = request.session["password"]
+            port = request.session["port"]
+            host = request.session["host"]
+            zone = request.session["zone"]
+            # use iget to transfer selected data object to local as a NamedTemporaryFile
+            irods_storage = IrodsStorage()
+            irods_storage.set_user_session(username=user, password=password, host=host, port=port, zone=zone)
+            tmpFile = irods_storage.download(irods_fname)
+            fname = os.path.basename(irods_fname.rstrip(os.sep))
+            res_files.append(UploadedFile(file=tmpFile, name=fname))
+
```

Comment:
Hmm.. okay, i did not realize the temporary location. That sounds fine, mostly. But, temporary file lifespans? I'm not sure about the lifespan. It may be deleted when it is closed, depending on the flags at creation time.

https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile